classes/factor.php

```diff
@@ -81,6 +81,7 @@ public function possible_states($user) {
      * Add an exemption for a user.
      *
      * @param \stdClass $user
+     * @param int $duration
```

Comment:
I believe the ?int should be included in PHPDoc for nullable fields

Reply:
Thanks, added.